### ISSUE TITLE
Unexpected nested containers behavior

### ISSUE THREAD

**Issue by samdark** [go-ozzo/ozzo-di#6]
https://github.com/go-ozzo/ozzo-di/blob/master/container.go#L222

If type isn't found, it's looked up in parent container and all of its dependencies are looked into parent container only. It doesn't look right to me if I have the following dependency tree:

```
Foo
  Bar
    Baz
Baz
```

Primary container has `Foo` and `Baz`. Parent container has `Bar` and `Baz`.

When building `Foo` from parent container it goes into parent container to get `Bar`. `Baz` is fetched from parent container despite the fact that we have it configured differently in the primary container.

**Comment by kolkov** [comment]
You're right — when resolving `Foo` from the primary container, the `build()` call jumps to the parent to get `Bar`, and then stays in the parent scope for all of `Bar`'s dependencies. So `Baz` is resolved from the parent even though the primary container has its own `Baz` registered. The lookup should walk back to the originating container before falling through to parents.

This is a real bug in the resolution logic. I'm now maintaining these packages and will look into fixing this.
